PurchasesCoreSwift/SubscriberAttributes/AttributionDataMigrator.swift

```diff
@@ -61,90 +61,91 @@ private extension AttributionDataMigrator {
         return networkSpecificSubscriberAttributes
     }
 
-    func convertMParticleAttribution(_ data: [String: Any]) -> [String: Any] {
-        var convertedAttribution: [String: Any] = [:]
-        if let value = data[AttributionKey.MParticle.id.rawValue] {
+    func convertMParticleAttribution(_ data: [String: Any]) -> [String: String] {
+        var convertedAttribution: [String: String] = [:]
+        if let value = data[AttributionKey.MParticle.id.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.mpParticleID] = value
         }
-        if let value = data[AttributionKey.networkID.rawValue] {
+        if let value = data[AttributionKey.networkID.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.mpParticleID] = value
         }
         return convertedAttribution
     }
 
-    func convertBranchAttribution(_ data: [String: Any]) -> [String: Any] {
-        var convertedAttribution: [String: Any] = [:]
-        if let value = data[AttributionKey.Branch.channel.rawValue] {
+    func convertBranchAttribution(_ data: [String: Any]) -> [String: String] {
+        var convertedAttribution: [String: String] = [:]
+        if let value = data[AttributionKey.Branch.channel.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.mediaSource] = value
         }
-        if let value = data[AttributionKey.Branch.campaign.rawValue] {
+        if let value = data[AttributionKey.Branch.campaign.rawValue] as? String {
             convertedAttribution[SpecialSubscriberAttributes.campaign] = value
         }
         return convertedAttribution
     }
 
     // swiftlint:disable cyclomatic_complexity
-    func convertAppsFlyerAttribution(_ data: [String: Any]) -> [String: Any] {
+    func convertAppsFlyerAttribution(_ data: [String: Any]) -> [String: String] {
         var fixedData = data
-        if let innerDataObject = fixedData[AttributionKey.AppsFlyer.dataKey.rawValue] as? [String: Any?] {
+        // TODO confirm casting all to strings makes sense
+        if let innerDataObject = fixedData[AttributionKey.AppsFlyer.dataKey.rawValue] as? [String: String?] {
```

Comment:
yeah, I think we don't have a choice but to do this optional casting for these. Developers directly send us the dicts coming in from their integrations, and we don't have a format enforced at the API level